frontend/src/components/resourceMap/graph/graphFiltering.ts

```diff
@@ -0,0 +1,90 @@
+import { getStatus } from '../nodes/KubeObjectStatus';
+import { GraphEdge, GraphNode } from './graphModel';
+
+export type GraphFilter =
+  | {
+      type: 'name';
+      query: string;
+    }
+  | {
+      type: 'hasErrors';
+    }
+  | {
+      type: 'namespace';
+      namespaces: Set<string>;
+    }
+  | {
+      type: 'related';
+      id: string;
+    }
+  | {
+      type: 'custom';
+      label: string;
+      filterFn: (node: GraphNode) => boolean;
+    };
+
+export function filterGraph(nodes: GraphNode[], edges: GraphEdge[], filters: GraphFilter[]) {
```

Comment:
It seems a lot of the functions in graph/ could be extracted into separate commits. The graphFiltering.ts should be in the same commit as the graphFiltering.test.ts

Reply:
In order to help with a good review now, and maintenance in the future.

There's many separate pieces which could be moved into atomic commits. With related changes grouped together - for example the tests in a separate commit mentioned above.

Reply:
isn't there a risk of losing context by introducing too much of splitting? having them in one commit makes it easier to understand how they relate directly. since this is an essential part of the map, wouldn't grouping it together provide a better view of the feature as a whole? at a certain point splitting everything into tiny commits might actually be harder to review since you have to switch contexts between different commits/PRs

Reply:
Having a commit with just a filterGraph and it's related tests in the same commit would be best IMHO.

Reply:
moved it to this PR #2499